src/main/java/org/folio/inventory/consortium/consumers/ConsortiumInstanceSharingHandler.java

```diff
@@ -0,0 +1,277 @@
+  }
+
+  @Override
+  public Future<String> handle(KafkaConsumerRecord<String, String> record) {
+    try {
+      Promise<String> promise = Promise.promise();
+      String consortiumId = record.key();
+      LOGGER.info("handle :: CONSORTIUM_INSTANCE_SHARING_INIT from consortiumId {}", consortiumId);
+      SharingInstance sharingInstance = Json.decodeValue(record.value(), SharingInstance.class);
+
+      String instanceId = sharingInstance.getInstanceIdentifier().toString();
+      Map<String, String> kafkaHeaders = KafkaHeaderUtils.kafkaHeadersToMap(record.headers());
+
+      LOGGER.info("Event CONSORTIUM_INSTANCE_SHARING_INIT has been received for instanceId: {}, sourceTenant: {}, targetTenant: {}",
+        instanceId, sharingInstance.getSourceTenantId(), sharingInstance.getTargetTenantId());
+      String tenantId = kafkaHeaders.get(OKAPI_TENANT_HEADER);
+
+      Context targetTenantContext = EventHandlingUtil.constructContext(sharingInstance.getTargetTenantId(),
+        kafkaHeaders.get(OKAPI_TOKEN_HEADER), kafkaHeaders.get(OKAPI_URL_HEADER));
+      InstanceCollection targetInstanceCollection = storage.getInstanceCollection(targetTenantContext);
+
+      Context sourceTenantContext = EventHandlingUtil.constructContext(sharingInstance.getSourceTenantId(),
+        kafkaHeaders.get(OKAPI_TOKEN_HEADER), kafkaHeaders.get(OKAPI_URL_HEADER));
+      InstanceCollection sourceInstanceCollection = storage.getInstanceCollection(sourceTenantContext);
+
+      LOGGER.info("handle :: checking is instance {} exists on tenant {}", instanceId, sharingInstance.getTargetTenantId());
+      getInstanceById(instanceId, sharingInstance.getTargetTenantId(), targetInstanceCollection)
+        .onFailure(failure -> {
+          if (failure.getClass().equals(NotFoundException.class)) {
+            LOGGER.info("handle :: instance {} not found on target tenant: {}", instanceId, sharingInstance.getTargetTenantId());
+            getInstanceById(instanceId, sharingInstance.getSourceTenantId(), sourceInstanceCollection)
+              .onSuccess(srcInstance -> {
+                if (srcInstance == null) {
+                  String errorMessage = format("instance %s not found on source tenant: %s", instanceId, sharingInstance.getSourceTenantId());
+                  sendCompleteEventToKafka(tenantId, sharingInstance, kafkaConfig, record.headers());
+                  promise.fail(errorMessage);
+                } else {
+                  LOGGER.info("handle :: Publishing instance {} with source {} from {} tenant to {} tenant", instanceId, srcInstance.getSource(), sharingInstance.getSourceTenantId(), sharingInstance.getTargetTenantId());
+                  if ("FOLIO".equals(srcInstance.getSource())) {
+                    publishInstanceToTenant(srcInstance, targetInstanceCollection)
+                      .onSuccess(publishedInstance -> {
+                        LOGGER.info("handle :: Updating source to 'CONSORTIUM-FOLIO' for Instance {}", instanceId);
```

Comment:
Every reference to CONSORTIUM-FOLIO should come from an enum or constant rather than a raw string.

Reply:
in the info message, indicate the the UUID provided is the instance id. Example `InstanceId=<UUID>`

Reply:
Completed.